tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
-            qml.tape.QuantumScript([qml.RY(0.5, 0)], [qml.expval(qml.Y(0))]),
+        tapes, fn = split_non_commuting(initial_tape, grouping_strategy=grouping_strategy)
+        assert len(tapes) == len(groups)
+        for tape, group in zip(tapes, groups):
+            assert tape.measurements == [qml.expval(term) for term in group]
+            assert tape.shots.total_shots == 100
+            assert tape.operations == [qml.X(0)]
+
+        assert qml.math.allclose(fn(results), expected_result)
+
+    @pytest.mark.parametrize("grouping_strategy", ["wires", "qwc"])
+    def test_single_group(self, grouping_strategy):
+        """Tests when all measurements can be taken at the same time."""
+
+        initial_tape = qml.tape.QuantumScript(
+            [qml.X(0)],
+            measurements=[qml.expval(qml.X(0)), qml.expval(qml.Y(1) + qml.Z(2))],
+            shots=100,
+        )
+        tapes, fn = split_non_commuting(initial_tape, grouping_strategy=grouping_strategy)
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [
+            qml.expval(qml.X(0)),
+            qml.expval(qml.Y(1)),
+            qml.expval(qml.Z(2)),
         ]
-        for actual_tape, expected_tape in zip(tapes, expected_tapes):
-            qml.assert_equal(actual_tape, expected_tape)
+        assert tapes[0].shots.total_shots == 100
+        assert tapes[0].operations == [qml.X(0)]
+        assert qml.math.allclose(fn([[0.1, 0.2, 0.3]]), [0.1, 0.2 + 0.3])
 
-        result = ([0.1, 0.2], 0.2, 0.3, 0.4)
-        assert fn(result) == ((0.1, 0.2, 0.2), (0.3, 0.4))
+    @pytest.mark.parametrize("grouping_strategy", ["wires", "qwc", None])
+    def test_single_observable(self, grouping_strategy):
+        """Tests a tape containing measurements of a single observable."""
 
-    @pytest.mark.parametrize(
-        "non_pauli_obs",
-        [
-            [
-                qml.Projector([0], wires=[1]),
-                qml.Projector([1, 1, 0, 1], wires=[0, 1]),
-            ],
-            [
-                qml.Hadamard(wires=[1]),
-                qml.Hadamard(wires=[0]) @ qml.PauliX(wires=[1]),
+        initial_tape = qml.tape.QuantumScript(
+            [qml.X(0)], measurements=[qml.expval(qml.Z(0)), qml.expval(0.5 * qml.Z(0))], shots=100
+        )
+        tapes, fn = split_non_commuting(initial_tape, grouping_strategy=grouping_strategy)
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [qml.expval(qml.Z(0))]
+        assert tapes[0].shots.total_shots == 100
+        assert tapes[0].operations == [qml.X(0)]
+        assert qml.math.allclose(fn([0.1]), [0.1, 0.05])
+
+    def test_mix_measurement_types_qwc(self):
+        """Tests multiple measurement types can be handled by qwc grouping"""
+
+        initial_tape = qml.tape.QuantumScript(
+            [qml.X(0)],
+            measurements=[
+                # [Z(0), X(0), Z(0), X(0), Y(1), Z(1), Z(0) @ Z(1), Z(1), Z(0)]
```

Comment:
Is this relevant to the statements below? Better to make it clear if so

Reply:
Added a comment.